packages/x-data-grid/src/components/base/GridOverlays.tsx

```diff
@@ -116,5 +116,10 @@ GridOverlayWrapper.propTypes = {
   // | To update them edit the TypeScript types and run "pnpm proptypes"  |
   // ----------------------------------------------------------------------
   loadingOverlayVariant: PropTypes.oneOf(['circular-progress', 'linear-progress', 'skeleton']),
-  overlayType: PropTypes.oneOf(['loadingOverlay', 'noResultsOverlay', 'noRowsOverlay']),
+  overlayType: PropTypes.oneOf([
+    'loadingOverlay',
+    'noResultsOverlay',
+    'noRowsOverlay',
+    'noColumnsOverlay',
```

Comment:
`pnpm proptypes` doesn't update this, and [it fails in React 18](https://app.circleci.com/pipelines/github/mui/mui-x/81326/workflows/6ebb4ce8-fcce-4f01-8dc5-2666ad0433fe/jobs/462937?invite=true#step-107-43221_169) (because it still runs proptypes checks) 🤷‍♂️